src/caja-desktop-window.c

```diff
@@ -230,15 +303,21 @@ unrealize (GtkWidget *widget)
     CajaDesktopWindow *window;
     CajaDesktopWindowPrivate *details;
     GdkWindow *root_window;
+    GdkDisplay *display;
 
     window = CAJA_DESKTOP_WINDOW (widget);
     details = window->details;
+    display = gdk_screen_get_display (gdk_screen_get_default());
```

Comment:
```suggestion
    display = gtk_widget_get_display (widget);
```
maybe?

Reply:
Didn't know we had that, will try it

Reply:
And in fact that works, saving us one variable (no matter how the source code calls it, the compiler still has to allocate it)

Reply:
Removing the check here, moving it to where this is called